### PR TITLE
[core] Allow LRU Evict to be specified from CLI

### PR DESCRIPTION
**Pull request by ijrsvt**
## Why are these changes needed?
This PR allows a user to launch a cluster from the CLI with LRU eviction enabled by calling `ray start --head --lru-evict`.
Before, the only way to run ray with LRU Eviction is through `ray.init(lru_evict=True)`.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Comment by pcmoritz** [comment]
Hmm, why are we exposing this, shouldn't it be removed?

**Comment by edoakes** [comment]
@pcmoritz why would this be removed? It's required for some workloads (e.g., jupyter notebook). Already exposed in ray.init, just never got added here.

File: python/ray/node.py [hunk 1]
```diff
@@ -93,6 +93,10 @@ def __init__(self,
                 "IP address when connecting to an existing raylet; i.e., when "
                 "head=False and connect_only=True.")
 
+        if ray_params._lru_evict:
+            assert (connect_only or
+                    head), "LRU Evict can only be passed into the head node."
```
Comment (stephanie-wang) [inline python/ray/node.py:98]: Why should it only be on the head node? I believe the worker nodes have to specify their own internal configs, and they should all match the head node's.

Comment (ijrsvt) [reply python/ray/node.py:98]: Oh, #8921 moves the internal config into GCS--so only the head node will be able to specify this.

Comment (stephanie-wang) [reply python/ray/node.py:98]: Oh gotcha, should we wait to merge this one then?

**Review by stephanie-wang — APPROVED** [review]
Thanks!

**Pull request merged**